repository/src/main/java/org/alfresco/repo/domain/node/AbstractNodeDAOImpl.java

```diff
@@ -563,6 +563,7 @@ public void beforeCommit(boolean readOnly)
             Long txnId = txn.getId();
             // Update it
             Long now = System.currentTimeMillis();
+            txn.setCommitTimeMs(now);

```

Comment:
I think this is done in updateTransaction method.

Reply:
not exactly, updateTransaction method creates a new object and uses it for the update, while I need to update the object in memory.
```
  @Override
    protected int updateTransaction(Long txnId, Long commitTimeMs)
    {
        TransactionEntity transaction = new TransactionEntity();
        transaction.setId(txnId);
        transaction.setCommitTimeMs(commitTimeMs);
        return template.update(UPDATE_TRANSACTION_COMMIT_TIME, transaction);
    }
```